packages/blockchain/src/index.ts

```diff
@@ -213,21 +214,26 @@ export default class Blockchain implements BlockchainInterface {
     this.db = opts.db ? opts.db : level()
     this.dbManager = new DBManager(this.db, this._common)
 
-    if (opts.consensus) {
-      this.consensus = new opts.consensus(
-        this.db,
-        this.dbManager,
-        this._common,
-        this._validateConsensus
-      )
-    } else if (this._common.consensusAlgorithm() === ConsensusAlgorithm.Clique) {
+    if (
+      opts.consensus === ConsensusAlgorithm.Casper ||
+      (!opts.consensus && this._common.consensusAlgorithm() === ConsensusAlgorithm.Casper)
+    ) {
+      this.consensus = new CasperConsensus(this.db)
+    } else if (
+      opts.consensus === ConsensusAlgorithm.Clique ||
+      (!opts.consensus && this._common.consensusAlgorithm() === ConsensusAlgorithm.Clique)
+    ) {
       this.consensus = new CliqueConsensus(
+        this,
         this.db,
         this.dbManager,
         this._common,
         this._validateConsensus
       )
-    } else if (this._common.consensusAlgorithm() === ConsensusAlgorithm.Ethash) {
+    } else if (
+      opts.consensus === ConsensusAlgorithm.Ethash ||
+      (!opts.consensus && this._common.consensusAlgorithm() === ConsensusAlgorithm.Ethash)
+    ) {
       this.consensus = new EthashConsensus(this.db)
     } else {
       throw new Error('Unable to initialize consensus class')
```

Comment:
Can you add a more explicit error here, "...consensus algorithm ${algorithm} not supported"?

Reply:
Updated in new PR